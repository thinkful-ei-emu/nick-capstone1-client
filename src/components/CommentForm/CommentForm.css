.comment-form-container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.comment-text {
  flex: .6;
  border-radius: 2px;
}

.comment-text-input {
  min-height: 100px;
}

.login-button {
  color: white;
  font-size: 18px;
  margin-left: 30px;
  min-height: 30px;
  width: 80px;
  background-color: rgb(207, 55, 55);
  border: 1px solid rgb(207, 55, 55);
  border-radius: 10px;
}

.comment-button {
  color: white;
  font-size: 18px;
  min-height: 30px;
  width: auto;
  background-color: rgb(207, 55, 55);
  border: 1px solid rgb(207, 55, 55);
  border-radius: 10px;
  margin: 4% 0;
}