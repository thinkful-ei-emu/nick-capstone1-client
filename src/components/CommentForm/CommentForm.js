import React from 'react';
import PostContext from '../../contexts/PostContext';
import PostApiService from '../../services/PostApiService';
import './CommentForm.css';

class CommentForm extends React.Component {
  static contextType = PostContext


  handleSubmit = ev => {
    ev.preventDefault()
    const {post} = this.context;
    const {text} = ev.target;

    PostApiService.postComment(post.id, text.value)
      .then(this.context.addComment)
      .then(() => {
        text.value = ''
      })
      .catch(this.context.setError)

  }

  render(){

    return (
      <form onSubmit={this.handleSubmit} >
        <div className='comment-form-container'>
          <textarea 
            name='text' 
            id='text' 
            placeholder='Leave your comment' 
            cols='30' 
            rows='3'
            className='comment-text comment-text-input'
            required>
          </textarea>
        </div>
        <button type='submit' className='comment-button'>Add Comment</button>
      </form>
    )
  }
}




export default CommentForm;