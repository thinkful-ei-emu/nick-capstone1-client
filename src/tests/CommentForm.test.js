import React from 'react';
import ReactDOM from 'react-dom';
import CommentForm from '../components/CommentForm/CommentForm';
import renderer from 'react-test-renderer';



it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<CommentForm />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders UI as expected', () => {
  const tree = renderer
      .create(<CommentForm />)
      .toJSON();
      expect(tree).toMatchSnapshot();
})