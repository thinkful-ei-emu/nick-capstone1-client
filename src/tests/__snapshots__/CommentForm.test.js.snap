// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="comment-form-container"
  >
    <textarea
      className="comment-text comment-text-input"
      cols="30"
      id="text"
      name="text"
      placeholder="Leave your comment"
      required={true}
      rows="3"
    />
  </div>
  <button
    className="comment-button"
    type="submit"
  >
    Add Comment
  </button>
</form>
`;
